### ISSUE TITLE
(QUESTION) Which model to use for multipas detections

### ISSUE THREAD

**Issue by MatheusCaltran** [jkjung-avt/tensorrt_demos#558]
Hello JK Jung

With your repositories I can learn more about AI, so I have a lot of consideration for you

I would like to know in your opinion which model I can use to detect several objects?(as shown in the image I have to do the cane detection in an agricultural machine)

![Snapshot_13](https://user-images.githubusercontent.com/88856093/171531560-89833c57-d74e-417a-b07e-a4158c719e20.png)
![Snapshot_81](https://user-images.githubusercontent.com/88856093/171531814-5734c34e-cd4f-474d-bd5f-092acf00799a.png)
![Snapshot_17](https://user-images.githubusercontent.com/88856093/171531882-ced07879-a217-417a-b553-0a121c4ae24f.png)

I'm thinking of using something like yolov4 tiny and leave the jetson nano inside the machine, or to use the yolov4 with a jetson xavier(I've done some training using SSD but I didn't have a good detection) Thanks in advance.

**Comment by MatheusCaltran** [comment]
thank you I hope one day to have a little of your knowledge, anything I repost the topic to ask about something else. But I would try to avoid it so as not to occupy your time.

**Comment by agjunyent** [comment]
If you only need to detect the canes (1 class), in this environment, yolov4-tiny-3l from darknet (https://github.com/AlexeyAB/darknet) will be more than sufficient, with good fps. If you increase the number of classes but maintain the "aspect ratio" of them (so all classes have similar sized boxes), yolov4-tiny-3l still is a very good candidate.

Crucial parts:
- Labelling, as @jkjung-avt said. I'd say that 50% of the success of a model, or even more, is training dataset labels quality
- Number of training samples. The more the better, but slower to label
- mAP requierements. You need to think that you will never have 100% accuracy / mAP.

**Comment by MatheusCaltran** [comment]
thank you very much, i will try with yolov4-tiny-3l sorry to take up your time

**Comment by MatheusCaltran** [comment]
thanks I will focus on using CSI, I will use ECAM-30 for now

**Comment by MatheusCaltran** [comment]
Hi @kjung-avt how are you?


we have already made several modifications to the machine, I am currently using jetson nano, e-cam30, yolov4-tiny

detection photo:
![11](https://user-images.githubusercontent.com/88856093/205403556-7c4ba98e-074a-4e1c-b728-1c0f8e67dfe9.jpg)
![19](https://user-images.githubusercontent.com/88856093/205403576-bc5cdbc2-9399-4924-8a9f-f390fb973ecc.jpg)
![50](https://user-images.githubusercontent.com/88856093/205403586-60f454a0-731d-4590-bd06-cb6d73b10e07.jpg)
![236](https://user-images.githubusercontent.com/88856093/205403645-707d43ed-10c7-4ba3-b517-aa862fca655a.jpg)

but now we do a test simulating reality and it wasn't very lucky because there was straw, but this will only happen in extreme cases because of the rain that makes the straw stick to the cane

example:
![guvcview_image-3](https://user-images.githubusercontent.com/88856093/205404030-25954444-fae0-4f2c-bf24-1ffab2a2b2be.jpg)

![guvcview_image-32](https://user-images.githubusercontent.com/88856093/205404052-7a1f287e-018c-4bb6-a5d1-42c10141b0d2.jpg)
![guvcview_image-47](https://user-images.githubusercontent.com/88856093/205404081-1746e6c0-4c04-4999-9973-f30b029d4562.jpg)
![guvcview_image-113](https://user-images.githubusercontent.com/88856093/205404135-0707458c-3b4a-4d24-a3c1-4dc4d7772a9a.jpg)

With all this I just want to know about your opinion what can I do to improve this? for me your opinion is very important, much of what I can do here was because of you.

**Comment by jkjung-avt** [comment]
Adding pictures with straws into your training set would definitely help.  I think you should do that.

I think a trained model should be able to detect sugarcanes which are partially covered by straws and still clearly visible.
